src/Generator/Generators/ExtensionMethods.cs

```diff
@@ -23,5 +24,35 @@ public static Interop.CallingConvention ToInteropCallConv(this CallingConvention
 
             return Interop.CallingConvention.Winapi;
         }
+
+        public static bool CheckClassIsStructible(Class @class, Driver Driver)
+        {
+            if (!@class.DeclaredStruct)
+                return false;
+
+            if (@class.IsValueType)
+                return true;
+            if (@class.IsInterface)
+                return false;
+            if (@class.IsStatic)
+                return false;
+            if (@class.IsAbstract)
+                return false;
+            if (@class.Methods.Any(m => m.IsStatic))
+                return false;
+            if (@class.Methods.Any(m => m.IsOperator))
+                return false;
+
+            var allTrUnits = Driver.ASTContext.TranslationUnits;
+            foreach (var trUnit in allTrUnits)
+            {
+                foreach (var cls in trUnit.Classes)
```

Comment:
Please verify once that it is correct way to loop through ALL the classes. I am not sure but have found no error so far.